File: src/main/java/searchalgorithms/linearsearch/java/LinearSearch.java
package searchalgorithms.linearsearch.java;

import org.jetbrains.annotations.NotNull;
import searchalgorithms.SearchAlgorithm;

public class LinearSearch extends SearchAlgorithm {
    @Override
    public int search(@NotNull int[] inputArray, int target) {
        for (int i=0;i< inputArray.length;i++) {
             if(inputArray[i] == target) return i;
        }
        return -1;
    }
}

File: src/main/java/searchalgorithms/linearsearch/kotlin/LinearSearch.kt
package searchalgorithms.linearsearch.kotlin

import searchalgorithms.SearchAlgorithm

class LinearSearch : SearchAlgorithm() {
    override fun search(inputArray: IntArray, target: Int): Int {
        for(i in inputArray.indices){
            if(inputArray[i] == target) return i;
        }
        return -1;
    }
}